src/lib/pickles/verify.ml

```diff
@@ -148,31 +148,49 @@ let verify_heterogenous (ts : Instance.t list) =
           in
           (absorb sponge, squeeze)
         in
-        ( absorb evals.ft_eval1 ;
-          let xs = Plonk_types.Evals.to_absorption_sequence evals.evals.evals in
-          let x1, x2 = evals.evals.public_input in
-          absorb x1 ;
-          absorb x2 ;
-          List.iter xs ~f:(fun (x1, x2) ->
-              Array.iter ~f:absorb x1 ; Array.iter ~f:absorb x2 ) ) ;
+        let old_bulletproof_challenges =
+          Vector.map ~f:Ipa.Step.compute_challenges
+            statement.pass_through.old_bulletproof_challenges
+        in
+        let sg_evals1, sg_evals2 =
+          let challenge_polys =
+            let vec =
+              Vector.map
+                ~f:(fun chals ->
+                  unstage (Wrap.challenge_polynomial (Vector.to_array chals)) )
+                old_bulletproof_challenges
+            in
+            fun pt -> Vector.map vec ~f:(fun f -> f pt)
+          in
+          (challenge_polys zeta, challenge_polys zetaw)
+        in
```

Comment:
this ^ computes the b_poly from the challenges (not sure why it's called `challenge_polys` and not `challenge_poly`, and then evaluate it at `zeta` and `zetaw`

Reply:
I think something like that would be clearer:

```ocaml
(* compute the b_poly polynomial from the challenges *)
let challenge_poly = 
  let f chals =
    unstage (Wrap.challenge_polynomial (Vector.to_array chals)) 
  in
  Vector.map ~f old_bulletproof_challenges
in

(* evaluate the b_poly polynomial at zeta and zeta * omega *)
let eval_b_poly pt = Vector.map challenge_poly ~f:(fun f -> f pt) in
let sg_evals1 = eval_b_poly zeta in
let sg_evals2 = eval_b_poly zetaw in
```